### PR TITLE
Refactor uses of react-native Text to ExpensifyText

### PR DESCRIPTION
**Pull request by puneetlath**
### Details

We recently [updated](https://github.com/Expensify/App/issues/3133) our custom `Text` component to be called `ExpensifyText` to more clearly distinguish it from the `react-native` component with the same name. We also [added a new rule](https://github.com/Expensify/App/pull/6583) to our ESLint config preventing the import of the default `Text` component from `react-native`. This is because we want people to use our custom component instead. 

There were a handful of instances of the default `react-native` component being used in our code-base. This PR updates those to use our custom component instead. The only file that still imports `Text` from `react-native` is our `ExpensifyText` component, which is built on top of the `react-native` one.  

cc @madmax330 since you reviewed the ESLint PR. 

### Fixed Issues

https://github.com/Expensify/App/issues/6585

### Tests

Open the page for each component that was changed and ensured the text looks the same/normal. 

### QA Steps

Check each component to make sure the text looks normal. Here's how to get to each component. 

CurrentWalletBalance and PaymentMethodList:
1. log into an account that has no payment methods added
2. tap the profile icon
3. Tap payments
4. Make sure the wallet balance (the big $ amount) and the rest of the text on the page look normal 

ExpensiPicker:
1. log into any account
2. tap the profile icon
3. tap profile
4. make sure the text in the "preferred pronounces" dropdown looks normal
 
InlineCodeBlock and PressableWithSecondaryInteraction:
1. Send someone a message that includes a url (i.e. `www.expensify.com`)
2. Make sure the text looks normal
3. Make sure you're able to long-press the text and have the "copy link to clipboard" menu show up
4. Send someone a message that includes a code block by wrapping the text in "`" characters. 
5. Make sure the text looks normal 

MarkerBadge:
1. Navigate to a chat that has more messages in the history than what shows on the screen
2. Scroll to the top
3. From another window, send a message from the other account that is in that chat
4. Notice the "1 new message" indicator now showing in the original account
5. Make sure the text in that indicator looks normal 

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

CurrentWalletBalance and PaymentMethodList:
image

ExpensiPicker:
Screen Shot 2021-12-08 at 6 28 49 PM

InlineCodeBlock and PressableWithSecondaryInteraction:
Screen Shot 2021-12-09 at 12 08 14 PM

MarkerBadge:
Screen Shot 2021-12-09 at 11 59 48 AM

#### Mobile Web

CurrentWalletBalance and PaymentMethodList:
Screen Shot 2021-12-09 at 12 35 22 PM

ExpensiPicker:
Screen Shot 2021-12-09 at 12 35 34 PM

InlineCodeBlock and PressableWithSecondaryInteraction:
Screen Shot 2021-12-09 at 12 35 46 PM

MarkerBadge:
Screen Shot 2021-12-09 at 12 36 01 PM

#### Desktop

CurrentWalletBalance and PaymentMethodList:
Screen Shot 2021-12-09 at 2 12 26 PM

ExpensiPicker:
Screen Shot 2021-12-09 at 2 12 38 PM

InlineCodeBlock and PressableWithSecondaryInteraction:
Screen Shot 2021-12-09 at 2 12 47 PM

MarkerBadge:
Screen Shot 2021-12-09 at 2 13 07 PM

#### iOS

CurrentWalletBalance and PaymentMethodList:
Screen Shot 2021-12-08 at 8 19 13 PM

ExpensiPicker:
Screen Shot 2021-12-08 at 8 19 41 PM

InlineCodeBlock and PressableWithSecondaryInteraction:
Screen Shot 2021-12-08 at 8 21 49 PM

MarkerBadge:
Screen Shot 2021-12-09 at 12 31 49 PM

#### Android

CurrentWalletBalance and PaymentMethodList:
Screen Shot 2021-12-08 at 7 43 26 PM

ExpensiPicker:
Screen Shot 2021-12-08 at 7 43 53 PM

InlineCodeBlock and PressableWithSecondaryInteraction:
Screen Shot 2021-12-08 at 7 44 45 PM

MarkerBadge:
Screen Shot 2021-12-09 at 12 26 27 PM

### CODE REVIEW

**Review by MonilBhavsar — COMMENTED** [review]
Looks good! Should we update the documentation to make sure we use ExpensifyText in future

**Comment by puneetlath** [comment]
I updated the ESLint config so that it'll throw an error if anyone uses `react-native Text`. We can definitely update the documentation as well. I don't see any obvious place to do that though. Anywhere in particular you recommend?

File: src/components/ExpensifyText.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import _ from 'underscore';
+
+// importing react-native Text as base for ExpensifyText component
```
Comment (puneetlath) [inline src/components/ExpensifyText.js:5]: @MonilBhavsar @madmax330 I was a bit on the fence about whether this comment was necessary in this scenario. It feels a bit obvious to me why we're ignoring the rule in this case and by adding this comment we end up having to add the blank line above which feels kinda ugly. What do y'all think?

Comment (MonilBhavsar) [reply src/components/ExpensifyText.js:5]: Yes, we can remove the comment as it doesn't explain much

**Comment by puneetlath** [comment]
@MonilBhavsar I removed the comment from `src/components/ExpensifyText.js`. Let me know if you're good with everything else!

**Comment by MonilBhavsar** [comment]
> I updated the ESLint config so that it'll throw an error if anyone uses react-native Text

That's great! Can you please link that PR.

> I don't see any obvious place to do that though. Anywhere in particular you recommend?

I don't see anywhere in documentation where we list components we use. So, all good! I was looking if we can update `STYLE.MD` for any occurrences of `Text`, But I saw you already updated it [here](https://github.com/Expensify/App/commit/7689aa0fb6b6edc2070994fa2852d23e1cbb6497). Thanks! 😄

**Comment by puneetlath** [comment]
Yep, here you go: https://github.com/Expensify/eslint-config-expensify/pull/38

**Review by MonilBhavsar — APPROVED** [review]
Thanks! Looks good

**Pull request merged**